compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -105,7 +105,15 @@ impl Context {
             .iter()
             .chain(function.returns())
             .filter(|id| function.dfg.get_numeric_constant(**id).is_none())
-            .copied();
+            .copied()
```

Comment:
I'm surprised that this needs both resolved and unresolved value ids, do the false positives still appear if you just use the resolved versions?

Reply:
I'll try